README.md

```diff
@@ -1,2 +1,101 @@
+  ```
+
+* Task-time metrics for each worker and task type used to generate the overall abnormality score (this can be presented as graphs with comparison to the historical models to make it human-readable):
+
+  ```
+  [(worker_id, worker_metric_0_task_0, ..., worker_metric_n_task_m), ...]
+  ```
+
+### Ona "Principal Surprise" Algorithm
+
+One basic approach to this worker abnormality problem is to compute the information gain (surprise) from a worker's task completion times based on historical models of task completion times.  Then, with a measure of surprise for each task type, we can determine the principal task types that differentiate workers and compute a weighted sum of the surprises as the overall abnormality.
+
+The idea is straightforward, but there are a few complications that need to be managed:
+
+* **Data varies greatly in quantity** - some task types will have much more historical data than others.  Also, in a week, for example, there is usually not enough data to directly build and compare a worker-specific model for a particular task type even with large data sets.
+* **Field managers usually have workers doing many different kinds of tasks** - ideally the worker ranking would smoothly handle a single task type, workers that do some tasks but not others, and/or workers doing any combination of heterogeneous tasks.
+* **There are lots of ways to build statistical models of task completion times** - the model building assumptions should ideally be a pluggable input to the overall algorithm.  Also, not all models will be equally "good" or predictive and our surprise metrics should reflect this.
+
+#### High-level overview
+
+Given the approach and requirements above, the PS algorithm works in the following way:
+
+1. ##### Compute task time surprise
+   1.  Load the historical task completion times for all workers, for each task type:
+      1. Generate a `lognorm` model of historical task completion times
+      2. Compute an `expected_(max_)err_pct` (percent) via resampling with a `kstest` metric
+   2. Load the analysis task completion times for all workers, for each worker, for each task type:
+      
+      1. Compute the sum and mean surprise (in bits) based on the historical task type model and expected error of that model.  
+         
+            High probability + low error ⇒ low surprise
+            Low probability + low error ⇒ high surprise
+            Low probability + high error ⇒ low surprise
+   
+2. ##### Project surprise metrics down to a ranking
+
+   1. Project the worker surprise metrics down to the most informative axes via a `PCA`
+   2. Compute a `multivariate_normal` distribution / mean in the projected space, rank workers by distance from mean
+      1. Also compute the `abs` individual contribution of each task-type surprise to the distance from mean for each worker to rank task-types by abnormality per-worker
+
+#### Example Use Case
+
+See `examples/transport.csv`
```

Comment:
This doesn't exist yet, but the test case examples used tasks like "bike", "walk", "drive", "fly" since they were good tasks with different overall durations.